### PR TITLE
Fix build and pipelines

### PR DESCRIPTION
**Pull request by beeender**
- Add VERSION file and unify the way of getting version string in the
  build scripts.
- Rewrite all pipelines.
- pljava_xxx_1.out files have been removed since they are not being used
  by anyone.
- pljava_test_optimizer.out has been removed. Since even when orca is
  enabled, the error message is the same as the planner:

```
< ERROR:  function cannot execute on a QE slice because it accesses relation "javatest.employees1"
< CONTEXT:  SQL statement "SELECT id, name, salary FROM employees1 WHERE salary > $1"
---
> ERROR:  query plan with multiple segworker groups is not supported
> HINT:  likely caused by a function that reads or modifies data in a distributed table
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -10,4 +11,5 @@ pljava-so/target/
 target/
 *.so
 *.zip
-concourse/secrets/
\ No newline at end of file
+concourse/secrets/
```
[the remark this answers is not in the document]
Comment (beeender) [reply .gitignore:14]: hmm , this should be deleted. will do

**Comment by liuxueyang** [comment]
There are some error logs in the `build_rhel7` [task](https://dev2.ci.gpdb.pivotal.io/teams/gp-extensions/pipelines/pr.pljava/jobs/build_test_rhel7/builds/2#L63a2e189:3517). I'm not sure whether it matters or not.

**Review by higuoxing — APPROVED** [review]
LGTM.

**Comment by beeender** [comment]
> There are some error logs in the build_rhel7 [task](https://dev2.ci.gpdb.pivotal.io/teams/gp-extensions/pipelines/pr.pljava/jobs/build_test_rhel7/builds/2#L63a2e189:3517). I'm not sure whether it matters or not.

Maven put an `ERROR` tag on the messages from stderr. And those are compile warnings actually. We can fix them in the following PRs.

**Review by liuxueyang — APPROVED** [review]

**Pull request merged**